sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/transform/BeamAggregationTransforms.java

```diff
@@ -223,13 +244,28 @@ public AggregationAccumulator createAccumulator() {
       }
       return initialAccu;
     }
+
     @Override
     public AggregationAccumulator addInput(AggregationAccumulator accumulator, BeamRecord input) {
       AggregationAccumulator deltaAcc = new AggregationAccumulator();
       for (int idx = 0; idx < aggregators.size(); ++idx) {
-        deltaAcc.accumulatorElements.add(
-            aggregators.get(idx).addInput(accumulator.accumulatorElements.get(idx),
-            sourceFieldExps.get(idx).evaluate(input, null).getValue()));
+        if (sourceFieldExps.get(idx) instanceof BeamSqlInputRefExpression) {
```

Comment:
Can you add some comments around here? Or extract functions with descriptive names? Otherwise this code is hard to trace through

Reply:
Done